src/fidesops/service/connectors/query_config.py

```diff
@@ -497,6 +520,42 @@ def get_formatted_query_string(
         return f'SELECT {field_list} FROM "{self.node.node.collection.name}" WHERE {" OR ".join(clauses)}'
 
 
+class BigQueryQueryConfig(QueryStringWithoutTuplesOverrideQueryConfig):
+    """
+    Generates SQL valid for BigQuery
+    """
+
+    def generate_update(
+        self, row: Row, policy: Policy, request: PrivacyRequest, client: Engine
+    ) -> Optional[Update]:
+        """
+        Using TextClause to insert 'None' values into BigQuery throws an exception, so we use update clause instead
+        """
+        update_value_map: Dict[str, Any] = self.update_value_map(row, policy, request)
+        non_empty_primary_keys: Dict[str, Field] = filter_nonempty_values(
+            {
+                fpath.string_path: fld.cast(row[fpath.string_path])
+                for fpath, fld in self.primary_key_field_paths.items()
+                if fpath.string_path in row
+            }
+        )
+
+        valid = len(non_empty_primary_keys) > 0
+        if not valid:
+            logger.warning(
+                f"There is not enough data to generate a valid update statement for {self.node.address}"
+            )
+            return None
+
+        table = Table(
+            self.node.address.collection, MetaData(bind=client), autoload=True
+        )
+        pk_clauses = [
+            getattr(table.c, k) == v for k, v in non_empty_primary_keys.items()
+        ]
+        return table.update().where(*pk_clauses).values(**update_value_map)
```

Comment:
Very nice solution to this @eastandwestwind, the SQLAlchemy table update method is much more readable